pkgs/by-name/mi/mill/package.nix

```
@@ -8,12 +8,20 @@
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "mill";
-  version = "0.12.5";
+  version = "0.12.8";
 
-  src = fetchurl {
-    url = "https://github.com/com-lihaoyi/mill/releases/download/${finalAttrs.version}/${finalAttrs.version}-assembly";
-    hash = "sha256-DHslQS/uzwbZVdATQY3pqQgM51W+26x2AckQnDPVoFc=";
-  };
+  src =
+    let
+      millUrl =
+        if lib.versionOlder finalAttrs.version "0.12.5" then
```

```suggestion
        if lib.versionOlder finalAttrs.version "0.12.6" then
```
Though I don't really think this is a good solution.
It would be better if the mill override in vyxal specified the old version and the old source link, in my opinion.

---

I agree, the mill package shouldn't need to handle different download locations based on version that adds unnecessary complexity.

I'll create a PR to try to fix vyxal's mill override to use the old source link for version 0.11.12